### PR TITLE
feat: Build script

### PR DESCRIPTION
**Pull request by jessicaschilling**
## In this PR

- Moves source-of-truth HTML and CSS files into new `src` directory
- <del>Adds new `dist` directory</del>
- Adds `build` script in `package.json` to do the following:
     - Wipe any existing files in `dist` directory
     - Write file `dist/dir-index.html` that includes a minified, inlined version of external CSS files from `src` directory (and removes their corresponding `<link>`s to stylesheets)
- Points code in `test` directory at the built `dist/dir-index.html`
- Updates readme accordingly

## <del>Important!</del>

<del>Closes https://github.com/ipfs/dir-index-html/issues/6, **BUT note that this will require updating go-ipfs itself to look for the new file names**. (@ribasushi and/or @lidel, is this something you're willing to address in a separate PR?)</del> 

(we kept the same filename, no changes to go-ipfs needed)

### CODE REVIEW

**Comment by jessicaschilling** [comment]
@lidel and @ribasushi, could you please weigh in? If you're OK with this approach, we may want to consider using a release based on this in https://github.com/ipfs/go-ipfs/pull/7536, for efficiency. Thank you!

**Comment by ribasushi** [comment]
@jessicaschilling I think this is great! One procedural nit: do we care how heavy the page is? If we do, perhaps we can use a real minifier in place of the basic `tr -d \"\t\n\r\"` ? One of these perhaps https://goalsmashers.github.io/css-minification-benchmark/

Note - if we do not care much about the pagesize and/or the source is already "pretty tight" - this comment should probably be disregarded.

A few stylistic nits remain, but if you decide to use a smarter minifier they are moot anyway.

**Comment by jessicaschilling** [comment]
@ribasushi Yep, the idea here was to keep the whole thing as lightweight as possible. We aren't using any other modules, so adding one just for this purpose didn't seem like the ideal way to go.

File: package.json [hunk 1]
```diff
@@ -11,8 +15,13 @@
...
   "language": "go",
   "license": "MIT",
-  "name": "dir-index-html",
   "releaseCmd": "git commit -a -m \"gx publish $VERSION\"",
-  "version": "1.0.4"
+  "scripts": {
+    "build": "npm run build:clean && npm run build:remove-style-links && npm run build:minify-wrap-css && npm run build:combine-html-css && npm run build:remove-unused",
+    "build:clean": "rm -rf ./dist/*",
+    "build:remove-style-links": "sed '/<link rel=\"stylesheet\"/d' ./src/dir-index.html > ./dist/base-html.html",
+    "build:minify-wrap-css": "(echo \"<style>\" && cat ./src/icons.css ./src/style.css | tr -d \"\t\n\r\" && echo \"</style>\") > ./dist/minified-wrapped-style.html",
```
Comment (ribasushi) [inline package.json:23]: > Yep, the idea here was to keep the whole thing as lightweight as possible

👍 

In that case my only nit is the following suggestion:

```suggestion
    "build:minify-wrap-css": "(echo \"<style>\" && cat ./src/icons.css ./src/style.css | tr -d \"\t\n\r\" && echo -e \"\\n</style>\") > ./dist/minified-wrapped-style.html",
```

Comment (jessicaschilling) [reply package.json:23]: oh! good catch 😊

**Comment by jessicaschilling** [comment]
Thanks, @ribasushi -- over to you, @lidel, for any thoughts 😊

**Comment by jessicaschilling** [comment]
@lidel Sounds like a good plan. Updated to remove `dist` directory and amend build script and test to look at the top-level dir instead. Back to you for the GitHub action finesse 😊  Thanks!

**Pull request merged**